kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java

```diff
@@ -110,6 +133,12 @@ private R applyWithSchema(R record) {
     }
     newValue.put(CdcConstants.COL_CDC, cdcMetadata);
 
+    if (kafkaAppender != null) {
+      if (record instanceof SinkRecord) {
```

Comment:
Don't love these casts but type `R` is erased at runtime and it's hard to get even via reflection. 

The better solution would be to separate this into concrete Source/SinkRecord classes but that would be a breaking change.

On a `Source` record there really is no point to injecting metadata.